src/main/java/org/reaktivity/specification/socks/internal/SocksFunctions.java

```diff
@@ -22,9 +22,25 @@
 import org.reaktivity.specification.socks.internal.types.control.SocksRouteExFW;
 import org.reaktivity.specification.socks.internal.types.stream.SocksBeginExFW;
 
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 public final class SocksFunctions
 {
     private static final int MAX_BUFFER_SIZE = 1024 * 8;
+    private static final Pattern IPV4_ADDRESS_PATTERN =
+        Pattern.compile("(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)" +
+                        ".(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)" +
+                        ".(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)" +
+                        ".(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)");
```

Comment:
```
$ jshell
|  Welcome to JShell -- Version 9.0.4
|  For an introduction type: /help intro

jshell> Pattern.compile("(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?).(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?).(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?).(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)")
$1 ==> (25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?).(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?).(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?).(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)

jshell> $1.matcher("192.168.0.1").matches()
$2 ==> true

jshell> $1.matcher("292.168.0.1").matches()
$3 ==> false

jshell> $1.matcher("0.0.0").matches()
$4 ==> false

jshell> $1.matcher("192x168x0x1").matches()
$5 ==> true
```
That last one should be `false`, due to `.` being a special character in Java patterns, but you need to match a literal `.` instead.

Reply:
done, update with `\\.`
```
jshell> $1.matcher("192x168x0x1").matches()
$5 ==> false
```